### PR TITLE
cmd: initialize node if not initialized

### PR DESCRIPTION
**Pull request by ryanchristo**
## Changes
- initialize node if the node has not been initialized
- refactor node initialization method (break up into smaller parts and add comments)

## Tests:
```
make gossamer
```
```
./bin/gossamer --datadir tmp/alice --key alice
```

### CODE REVIEW

**Comment by noot** [comment]
this seems okay, maybe we can discuss it on the call tomorrow?

**Comment by ryanchristo** [comment]
@noot ok, I updated it with a bit of refactoring the initialization method, let's discuss tomorrow

File: cmd/gossamer/main.go [hunk 1]
```diff
@@ -128,18 +130,61 @@ func gossamer(ctx *cli.Context) error {
...
+
+	// start node
 	node.Start()
 
 	return nil
 }
+
+// nodeInitialized returns true if, within the configured data directory for the
+// node, the state database has been created and the gensis data has been loaded
+func nodeInitialized(ctx *cli.Context) bool {
```
Comment (ryanchristo) [inline cmd/gossamer/main.go]: is there a better way to check if a node is initialized without being too costly (ie, spinning up the state service)?

File: cmd/gossamer/genesis.go [hunk 2]
```diff
@@ -134,29 +149,54 @@ func initializeGenesisState(gen *genesis.Genesis) (*trie.Trie, *types.Header, er
...
-	if file := ctx.String(GenesisFlag.Name); file != "" {
-		return file
-	} else if file := ctx.GlobalString(GenesisFlag.Name); file != "" {
-		return file
-	} else if name := ctx.GlobalString(NodeFlag.Name); name != "" {
-		return path.Join("node", name, "genesis.json")
-	} else {
-		return gssmr.DefaultGenesisPath
+// initializeTrieDatabase initializes and sets the trie database and then stores
+// the encoded trie, the genesis hash, and the gensis data in the trie database
```
Comment (noot) [inline cmd/gossamer/genesis.go]: genesis*

**Pull request merged**